File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.js
function solution(n) {
    var answer = Array.from(Array(n), () => Array(n).fill(null));
    let num = 1
    let [startRow, endRow, startCol, endCol] = [0, n - 1, 0, n - 1]
    
    while(num <= Math.pow(n, 2)) {
        for(let j = startCol; j <= endCol; j++) answer[startRow][j] = num++
        startRow += 1
        
        for(let i = startRow; i <= endRow; i++) answer[i][endCol] = num++
        endCol -= 1
        
        for(let j = endCol; j>= startCol; j--) answer[endRow][j] = num++
        endRow -= 1
        
        for(let i = endRow; i >= startRow; i--) answer[i][startCol] = num++
        startCol += 1
    }
    
    return answer;
}

File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.java
class Solution {
    public int[][] solution(int n) {
        int[][] answer = new int[n][n];
        
        int i = 1, startRow = 0, endRow = n - 1, startCol = 0, endCol = n-1;
        while(i <= n*n){
            // 왼위 -> 오위
            for(int j = startCol; j <= endCol; j++) answer[startRow][j] = i++;
            startRow++;
            
            // 오위 -> 오아래
            for(int j = startRow; j <= endRow; j++) answer[j][endCol] = i++;
            endCol--;
            
            // 오아래 -> 왼아래
            for(int j = endCol; j >= startCol; j--) answer[endRow][j] = i++;
            endRow--;
            
            // 왼아래 -> 왼위
            for(int j = endRow; j >= startRow; j--) answer[j][startCol] = i++;
            startCol++;
        }
        return answer;
    }
}